### DISCUSSION TITLE
findUnique with nested queries on compound unique keys

### DISCUSSION THREAD

**Discussion by NicolasEspiau-stilll** [prisma/prisma#21755] [Q&A] [answered]
### Question

Hello folks,

I'm starting to use Prisma, I'm a beginner.

I developing using VS Code, with node 18 and Typescript 3.7.4, and prisma 5.5.2.

I've designed a pretty standard schema with foreign keys included in compound unique constraints:
```prisma
model Workflow {
  id                  Int         @id @default(autoincrement())
  name            String    @unique
  GroupTypes GroupType[]
}

model GroupType {
  id                        Int           @id @default(autoincrement())
  workflow_id        Int
  workflow             Workflow      @relation(fields: [workflow_id], references: [id])
  name                  String
  description          String?
  GroupStatuses   GroupStatus[]
  Groups               Group[]

  @@unique([workflow_id, name])
}
```

From there, I need to perform a findUnique query but I don't know the `workflow_id` so I would like to use nested query like the following:
```ts
 const group_type = prismaClient.groupType.findUnique({
      where: {
        workflow_id_name: {
          workflow: {
            name: "Workflow1"
          },
          name: "Group1-1"
        }
      }
    })
```

But this is rejected with the error:
```bash
Type '{ workflow: { name: string; }; name: string; }' is not assignable to type 'GroupTypeWorkflow_idNameCompoundUniqueInput | (number & GroupTypeWorkflow_idNameCompoundUniqueInput) | undefined'.
  Object literal may only specify known properties, and 'workflow' does not exist in type 'GroupTypeWorkflow_idNameCompoundUniqueInput'.
```

So, what am I doing wrong?
The Typescript autocompletion does not provide other choice than `workflow_id` (which would need a prior request) and `name` and TS wont accept anything else.

Thanks in advance for you help.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @NicolasEspiau-stilll 👋 

Thank you for raising this question.

Your compound key is supposed to search on the `workflow_id` and `name` fields like this:
```js
 workflow_id_name: {
    workflow_id: workflow.id, 
    name: "Group1-1",
 },
```

but since you don't know the `workflow_id`, you can perform two queries. 

1. Find the workflow_id based on the workflow name
```js
const workflow = await prismaClient.workflow.findUnique({
  where: {
    name: "Workflow1",
  },
```
2. Use the obtained workflow_id in the findUnique query for GroupType
```js
const groupType = await prismaClient.groupType.findUnique({
  where: {
    workflow_id_name: {
      workflow_id: workflow.workflow_id, // Use the obtained workflow_id
      name: "Group1-1",
    },
  },
});
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by NicolasEspiau-stilll** [reply]
that's what I've done in the meantime to make my script work, but I thought it was possible to do it with a nested query.

Thank you for your answer!
